src/main/scala/com.snowplowanalytics/iglu.schemaddl/migrations/SchemaList.scala

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (c) 2014-2019 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.iglu.schemaddl.migrations
+
+// cats
+import cats.Monad
+import cats.data._
+import cats.implicits._
+import cats.kernel.Order
+
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaList => SchemaKeyList, SchemaMap, SchemaVer}
+
+import com.snowplowanalytics.iglu.schemaddl.{IgluSchema, ModelGroup, VersionTree}
+import SchemaList.FullListCreationError._
+
+/**
+  * Always belong to the same vendor/name/model triple.
+  * Schemas in the list will be in correct order.
+  */
+sealed trait SchemaList extends Product with Serializable
+
+object SchemaList {
+
+  /**
+    * Always belong to the same vendor/name/model triple,
+    * always have at least two elements: head (initial x-0-0 schema) and latest schema
+    */
+  case class SchemaListFull private(schemas: NonEmptyList[IgluSchema]) extends SchemaList {
```

Comment:
That's a common convention in Scala to refer to ADTs as to `Root.Subtype`. Where `Root` is an ADT name and `Subtype` is data constructor. We can turn `SchemaList.SchemaListFull` into `SchemaList.Full`, which IMO much more concise and readable.

Reply:
Same for all other data constructors.